### ISSUE TITLE
Kompose fails on version 3 with deploy-resources-limits (panic: runtime error: invalid memory address or nil pointer dereference)

### ISSUE THREAD

**Issue by mvreg** [kubernetes/kompose#959]
Using kompose version 1.10.0 (8bb0907) on Ubuntu 16.04 and converting using 
```
kompose convert -f b.yaml
```

-----

b.yaml:

```
services:
  haproxydocker:
    command: haproxy -f /etc/haproxy/haproxy_dsp.cfg
    environment:
      BACKEND_1_IP: 192.168.1.105
    image: haproxy
    network_mode: host
    restart: always
    deploy:
      resources:
        limits:
          memory: 256m
version: "3"
```
---------
I get following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1622d57]

goroutine 1 [running]:
github.com/kubernetes/kompose/pkg/loader/compose.dockerComposeToKomposeMapping(0xc420010600, 0x1c, 0xc420133c20, 0x1, 0x1, 0xc4201a5e60)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/loader/compose/v3.go:267 +0xfe7
github.com/kubernetes/kompose/pkg/loader/compose.parseV3(0xc42052db70, 0x1, 0x1, 0x1, 0x1, 0x0, 0x30, 0x1bec580)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/loader/compose/v3.go:113 +0x22b
github.com/kubernetes/kompose/pkg/loader/compose.(*Compose).LoadFile(0x2b1f0d8, 0xc42052db70, 0x1, 0x1, 0xc4201a5830, 0x0, 0x1a, 0x7f6a9adfcd50, 0x454d10)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/loader/compose/compose.go:186 +0x4d2
github.com/kubernetes/kompose/pkg/app.Convert(0x100000100, 0x0, 0x0, 0x0, 0x0, 0x1cac29c, 0x4, 0x0, 0x1cc0501, 0x15, ...)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/app/app.go:232 +0x145
github.com/kubernetes/kompose/cmd.glob..func3(0x2aec8e0, 0xc42045f900, 0x0, 0x2)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/cmd/convert.go:95 +0x4d
github.com/kubernetes/kompose/vendor/github.com/spf13/cobra.(*Command).execute(0x2aec8e0, 0xc42045f8c0, 0x2, 0x2, 0x2aec8e0, 0xc42045f8c0)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/vendor/github.com/spf13/cobra/command.go:704 +0x2c6
github.com/kubernetes/kompose/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2aecd20, 0x2aed160, 0x18a1b60, 0x2aece60)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/vendor/github.com/spf13/cobra/command.go:785 +0x30e
github.com/kubernetes/kompose/vendor/github.com/spf13/cobra.(*Command).Execute(0x2aecd20, 0x0, 0xc420275f48)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/vendor/github.com/spf13/cobra/command.go:738 +0x2b
github.com/kubernetes/kompose/cmd.Execute()
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/cmd/root.go:92 +0x31
main.main()
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/main.go:22 +0x20
```
-----

If I comment out deploy section:
```
    # deploy:
      # resources:
        # limits:
          # memory: 256m
```
it works OK and creates deployment and service. yaml

To be sure that this is docker-compose.yaml (b.yaml) is valid synthax I also checked with
docker-compose -f b.yaml config
and it works OK (used docker-compose version 1.19.0, build 9e633ef)

**Comment by mvreg** [comment]
Narrowd down the problem to the point that when I add services.deploy.resources.reservations it works OK.

```
services:
  haproxydocker:
    command: haproxy -f /etc/haproxy/haproxy_dsp.cfg
    environment:
      BACKEND_1_IP: 192.168.1.105
    image: haproxy
    network_mode: host
    restart: always
    deploy:
      resources:
        limits:
          memory: 256m
          cpus: '0.01'
        reservations:
          cpus: '0.01'
          memory: 128M
version: "3"
```

**Comment by hangyan** [comment]
I will take a look

**Comment by malcolmgreaves** [comment]
Can confirm this behavior on `kompose` version 1.10.0. Furthermore, I have found out that *the only way this works* is if (1) both `limits` and `reservations` are defined and (2) both sections have `cpus` and `memory` defined. Any other combination will not work correctly.

Most combinations result in SIGSEV. However, I noticed that if one defines everything _except_ for the `cpus` key for `limits`, then one gets this error message:

```
FATA Unable to convert cpu limits resources value: strconv.ParseFloat: parsing "": invalid syntax
```

This leads me to believe that the problem is that, somewhere in the `kompose` code, the function responsible for parsing this `resources` section incorrectly assumes that there is always a `cpus` option specified.

Hope this information helps! A fix would be greatly appreciated.
